allennlp/modules/seq2seq_encoders/compose_encoder.py

```diff
@@ -0,0 +1,76 @@
+from overrides import overrides
+import torch
+from typing import List
+
+from allennlp.modules.seq2seq_encoders.seq2seq_encoder import Seq2SeqEncoder
+
+
+@Seq2SeqEncoder.register("compose")
+class ComposeEncoder(Seq2SeqEncoder):
+
+    """This class can be used to compose several encoders in sequence.
+
+    Among other things, this can be used to add a "pre-contextualizer" before a Seq2SeqEncoder.
+
+    Parameters
+    ----------
+    encoders : ``List[Seq2SeqEncoder]``, required.
+        A non-empty list of encoders to compose. The encoders must match in bidirectionality.
+    """
+
+    def __init__(self, encoders: List[Seq2SeqEncoder]):
+        super().__init__()
+        self.encoders = encoders
+        for idx, encoder in enumerate(encoders):
+            self.add_module("encoder%d" % idx, encoder)
+
+        # Compute bidirectionality.
+        all_bidirectional = all(encoder.is_bidirectional() for encoder in encoders)
+        any_bidirectional = any(encoder.is_bidirectional() for encoder in encoders)
+        self.bidirectional = all_bidirectional
+
+        if all_bidirectional != any_bidirectional:
+            raise ValueError("All encoders need to match in bidirectionality.")
+
+        if len(self.encoders) < 1:
+            raise ValueError("Need at least one encoder.")
```

Comment:
It would be neat if this just does pass-through in the case of zero encoders, but I assume that doesn't work because then you can't compute input and output dimensionality?

Reply:
Yeah, exactly. The pass through requires an “embedding dim” in order to fulfill that part of the API.